### PR TITLE
Configurable chart line width

### LINKED ISSUE #2770: Configurable line width for line charts
Sometimes it's quite difficult to read the line charts precisely when there are multiple overlapping series. 
image

I'd suggest, to make the line with configurable like this:
image

The topic is also mentioned in the [forum](https://forum.portfolio-performance.info/t/liniendicke-in-diagrammen/20288)

### PR DESCRIPTION
**Pull request by eztam-**
Closes #2770

### CODE REVIEW

**Comment by Nirus2000** [comment]
Hello @eztam- 
I think that "Linien Dicke" is not the right word for it.
`Linien Dicke` -> Linienbreite oder Linienstärke
Please change the translations and add them in all other languages.

**As a suggestion I would take the following translations:**
[DEFAULT] Line width
[CS] Šířka čáry
[DE] Linienbreite 
[ES] Ancho de línea
[FR] Largeur de ligne
[IT] Larghezza della linea
[NL] Lijnbreedte
[PT] Largura de linha
[RU] Ширина линии

Greets
Alex

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesChartLegend.java [hunk 1]
```diff
@@ -242,6 +245,19 @@ public void run()
...
                 actionShowArea.setChecked(series.isShowArea());
                 manager.add(actionShowArea);
+                
+                MenuManager lineWidth = new MenuManager(Messages.ChartSeriesPickerLineWidth);
+                IntStream.range(1, 3).forEach(i ->{
+                    Action action = new SimpleAction(i + " px", a -> { //$NON-NLS-1$
+                        series.setLineWidth(i);
+                        configurator.fireUpdate();
+                    });
+                    action.setChecked(i==series.getLineWidth());
```
Comment (Nirus2000) [inline name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesChartLegend.java:255]: `action.setChecked(i == series.getLineWidth());` :wink:

Comment (eztam-) [reply name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesChartLegend.java]: Thanks for the hints. I just updated all your points. Somehow the ResourceBundle Editor has also automatically adjusted the encoding of one other property. But that should be OK.

**Comment by eztam-** [comment]
Hi @Nirus2000,
thanks for the hints! I just updated all your points. Somehow the ResourceBundle Editor has also automatically adjusted the encoding of one other property. But that should be OK.

Regards,
Matthias

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesSerializer.java [hunk 2]
```diff
@@ -49,12 +49,16 @@ private void load(DataSeriesSet set, String config, List<DataSeries> series)
             {
                 series.add(s);
 
-                if (data.length == 4)
+                if (data.length >=4 && data.length <=5)
```
Comment (OnkelDok) [inline name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesSerializer.java:52]: Shouldn't the condition ``data.length >= 4`` be sufficiant?
Then it's quite easier to add more data items in the future without checking the old conditions. The condition can remain as long as data fields are added.

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesSerializer.java [hunk 3]
```diff
@@ -49,12 +49,16 @@ private void load(DataSeriesSet set, String config, List<DataSeries> series)
...
                 {
                     s.setColor(ColorConversion.hex2RGB(data[1]));
                     s.setLineStyle(LineStyle.valueOf(data[2]));
                     s.setShowArea(Boolean.parseBoolean(data[3]));
                 }
+                if(data.length == 5)
```
Comment (OnkelDok) [inline name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesSerializer.java:58]: Here I would suggest to change the condition to ``data.length >= 5``. This has the same advantage as described above:
Then it's quite easier to add more data items in the future without checking the old conditions. The condition can remain as long as data fields are added.

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesSerializer.java (hunk 2 above)
Comment (eztam-) [reply name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesSerializer.java]: Thanks for the hint. I agree.

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/AbstractChartSeriesBuilder.java [hunk 4]
```diff
@@ -41,11 +41,12 @@ protected void configure(DataSeries series, ILineSeries lineSeries)
...
+        lineSeries.setLineWidth(series.getLineWidth());
     }
 
     protected void configure(DataSeries series, IBarSeries barSeries)
     {
         barSeries.setBarPadding(50);
         barSeries.setBarColor(resources.createColor(series.getColor()));
     }
-}
+}
```
Comment (OnkelDok) [inline name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/AbstractChartSeriesBuilder.java:52]: github shows following hint here: "No newline at the end of file"

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeries.java [hunk 5]
```diff
@@ -259,4 +270,4 @@ public String toString()
     {
         return getSearchLabel() + " [" + getUUID() + "]"; //$NON-NLS-1$ //$NON-NLS-2$
     }
-}
+}
```
Comment (OnkelDok) [inline name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeries.java:273]: github shows following hint here: "No newline at the end of file"

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesChartLegend.java [hunk 6]
```diff
@@ -502,4 +516,4 @@ private void overflowMenuAboutToShow(IMenuManager manager)
             }
         }
     }
-}
+}
```
Comment (OnkelDok) [inline name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesChartLegend.java:519]: github shows following hint here: "No newline at the end of file"

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesSerializer.java [hunk 7]
```diff
@@ -69,9 +73,10 @@ public String toString(List<DataSeries> series)
...
             buf.append(s.getLineStyle().name()).append(';');
-            buf.append(s.isShowArea());
+            buf.append(s.isShowArea()).append(';');
+            buf.append(s.getLineWidth());
         }
         return buf.toString();
     }
 
-}
+}
```
Comment (OnkelDok) [inline name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dataseries/DataSeriesSerializer.java:82]: github shows following hint here: "No newline at the end of file"

**Review by OnkelDok — COMMENTED** [review]
I just noticed that github is displaying following hint at the end of each java-file:
grafik

**Comment by Nirus2000** [comment]
Hallo @eztam- 
Can you check in your Eclips installation if you have installed CheckStyle? (Eclipse Checkstyle Plug-in)
If yes, then "CTRL + Shift + F" for the formatting of the source code.

Greets
Alex

**Comment by eztam-** [comment]
@Nirus2000  Thanks, I really appreciate this kind of suggestions! I haven't used Eclipse since many years and have to get familiar with it again. Will install the plugin 👍

**Comment by buchen** [comment]
Thanks. Nice work. (I allowed also for 3px line width - sometime I rather want to highlight one trend)

**Pull request closed without merging**